### PR TITLE
.js file imports and schema loading as statefield

### PR DESCRIPTION
**Pull request by acao**
**this introduces a breaking change!**
with this change, you'll be able to provide schema this way:
```ts

import { jsonSchema, updateSchema } from 'codemirror-json-schema'
const state = EditorState.create({ 
// no need to pass schema here anymore
extensions: [jsonSchema()] 
})

const view = new EditorView({ state, parent: selector })

updateSchema(view, schema)
```

I'm going to do one more pass to change the signature of the linter extensions, where they are exported with the `needsRefresh` logic

### CODE REVIEW

File: src/json-completion.ts [hunk 1]
```diff
@@ -681,7 +687,7 @@ export class JSONCompletion {
     schema: JSONSchema7,
     ctx: CompletionContext
   ): JSONSchema7Definition[] {
-    const draft = new Draft07(this.schema);
+    const draft = new Draft07(this.schema!);
```
Comment (imolorhe) [inline src/json-completion.ts:690]: I'm always wary of using this ! in the code

File: src/json5-validation.ts [hunk 2]
```diff
@@ -1,21 +1,22 @@
 import { EditorView } from "@codemirror/view";
-import { JSONValidation, type JSONValidationOptions } from "./json-validation";
-import type { JSONSchema7 } from "json-schema";
-import { parseJSON5DocumentState } from "./utils/parseJSON5Document";
+import { linter } from "@codemirror/lint";
+import {
+  JSONValidation,
+  handleRefresh,
+  type JSONValidationOptions,
```
Comment (imolorhe) [inline src/json5-validation.ts:6]: Did you have to manually update all the imports to use the .js extension?

**Review by imolorhe — COMMENTED** [review]
Looking good!

File: src/json5-validation.ts (hunk 2 above)
Comment (acao) [reply src/json5-validation.ts:6]: yes, this is part of adopting the new node module resolution spec!

**Comment by acao** [comment]
@imolorhe i forgot to note that this is a breaking change before, and there is no change to the readme yet for this PR, danger! I will fix this haha

**Comment by acao** [comment]
@imolorhe this introduces a breaking change for anyone using the "custom" usage approach, so not bad news at all!

**Pull request merged**